### ISSUE TITLE
Hydration bug with browser extensions

### ISSUE THREAD

**Issue by probablykasper** [sveltejs/kit#4508]
### Describe the bug

https://limey.io is using SvelteKit, and many users are reporting that the site isn't loading properly. It seems that hydration isn't happening. In some very rare occurrences, it does hydrate.

The issue only happens with certain browser extensions, for example the [BeaNote](https://chrome.google.com/webstore/detail/beanote-note-taking-on-we/nikccehomlnjkmgmhnieecolhgdafajb) extension.

I've reproduced it on Netlify and CloudFlare Pages, but not locally.

### Reproduction

Repo: https://github.com/probablykasper/sveltekit-hydration-bug

1. Deploy to Netlify or CloudFlare Pages
2. Install the [BeaNote](https://chrome.google.com/webstore/detail/beanote-note-taking-on-we/nikccehomlnjkmgmhnieecolhgdafajb?hl=en)
3. Visit the deployed site

### System Info

```shell
System:
    OS: macOS 10.15.7
    CPU: (8) x64 Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz
    Memory: 396.21 MB / 32.00 GB
    Shell: 3.2.2 - /usr/local/bin/fish
  Binaries:
    Node: 16.13.1 - /var/folders/gs/3qcm1mc918s08sxvkj6zj2rr0000gp/T/fnm_multishells/39265_1649091675026/bin/node
    Yarn: 1.22.17 - /var/folders/gs/3qcm1mc918s08sxvkj6zj2rr0000gp/T/fnm_multishells/39265_1649091675026/bin/yarn
    npm: 8.1.2 - /var/folders/gs/3qcm1mc918s08sxvkj6zj2rr0000gp/T/fnm_multishells/39265_1649091675026/bin/npm
  Browsers:
    Brave Browser: 97.1.34.81
    Chrome: 99.0.4844.84
    Firefox: 98.0
    Safari: 15.4
  npmPackages:
    @sveltejs/adapter-static: ^1.0.0-next.29 => 1.0.0-next.29 
    @sveltejs/kit: ^1.0.0-next.303 => 1.0.0-next.304 
    svelte: ^3.46.6 => 3.46.6
```

### Severity

blocking all usage of SvelteKit

**Comment by probablykasper** [comment]
It seems like having `app.html` body like this causes it:
```
  <body>
    %svelte.body%
  </body>
```
Switching to this fixes it:
```
  <body>
    <div>%svelte.body%</div>
  </body>
```

I couldn't find any documentation stating that you should wrap `%svelte.body%` in a tag though. In fact, the Sapper migration guide specifically states "The `<div id="sapper">` is no longer necessary".

**Comment by Rich-Harris** [comment]
Yeah, the default `app.html` files in the project templates are wrapped for this reason. The best we can do is figure out how to document it so that people don't run into this issue

**Comment by PH4NTOMiki** [comment]
@probablykasper kinda related, you should also probably disable Rocket Loader (in the Cloudflare dashboard) for your page since it delays loading of the JS(at least until they rocket loader library loads) and since you are using SPA, you need JS for the page to be able to load, so it's needless, and could actually break the page

**Comment by probablykasper** [comment]
@PH4NTOMiki Thanks for the recommendation!

**Comment by geoffrich** [comment]
Closed by #6245
